airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_budget_stream.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+from airbyte_cdk.models import Level, SyncMode
+from airbyte_cdk.test.mock_http import HttpMocker
+from test_bulk_stream import TestBulkStream
+
+
+class TestBudgetStream(TestBulkStream):
+    stream_name = "budget"
+    account_id = "180535609"
+
+    @HttpMocker()
+    def test_return_records_from_given_csv_file(self, http_mocker: HttpMocker):
+        self.auth_client(http_mocker)
+        output = self.read_stream(self.stream_name, SyncMode.full_refresh, self._config, [], "budget")
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_return_logged_info_for_empty_csv_file(self, http_mocker: HttpMocker):
+        self.auth_client(http_mocker)
+        output = self.read_stream(self.stream_name, SyncMode.full_refresh, self._config, [], "budget_empty")
+        assert len(output.records) == 0
+        assert "Empty data received. No columns to parse from file" in [log.log.message for log in output.logs]
+
+    @HttpMocker()
+    def test_return_logged_fatal_for_ioerror(self, http_mocker: HttpMocker):
+        self.auth_client(http_mocker)
+        output = self.read_stream(self.stream_name, SyncMode.full_refresh, self._config, [], stream_data_file=None)
+        fatal_log = [log.log for log in output.logs if log.log.level == Level.FATAL]
+        assert fatal_log
+        assert "The IO/Error occurred while reading tmp data. Called:" in fatal_log[0].message
+        assert len(output.records) == 0
+
+    @HttpMocker()
+    def test_transform_records(self, http_mocker: HttpMocker):
+        self.auth_client(http_mocker)
+        output = self.read_stream(self.stream_name, SyncMode.full_refresh, self._config, [], "budget")
+        for record in output.records:
+            assert "Account Id" in record.record.data.keys()
+            assert isinstance(record.record.data["Account Id"], int)
+
+    @HttpMocker()
+    def test_reads_with_chunks_from_given_csv_file_with_big_amount_of_data(self, http_mocker: HttpMocker):
```

Comment:
What is the different behavior between this test and `test_return_records_from_given_csv_file`? They seem identical based on this code so something might need to be more explicit here

Reply:
in these case with big amount of data we test read_with_chunks() behavior, chunk size is 1024 by default

Reply:
In my mind, this should be a unit test and not an integration test as the behavior is not different from the other test. Is there a reason why we can't unit test this?

Reply:
it tests in unit test also

Reply:
So let's not duplicate this test. Integration tests are to ensure that the component are interacting properly. If we already have one test that validates that the source can interact properly with the CSV parser unit, we don't need another one except if there is something very dangerous or specific to be highlighted

Reply:
I will remove test with big amount of data